From 13021115edea66e49b974166e332e1dd09851f20 Mon Sep 17 00:00:00 2001
From: Tom Marshall <tdm@cyngn.com>
Date: Fri, 30 Oct 2015 06:21:38 -0700
Subject: [PATCH 1/3] minizip: Fix build under Android 6.0 and higher

See bionic change Ic3cb409aae6713f4b345de954bcc4241fcd969ec

Change-Id: Ifb513b2c8e78c0f7035da39b475672d0d8cd37d4
---
 src/contrib/minizip/ioapi.h   | 2 +-
 src/contrib/minizip/miniunz.c | 2 +-
 src/contrib/minizip/minizip.c | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/contrib/minizip/ioapi.h b/src/contrib/minizip/ioapi.h
index 8dcbdb0..3b7e4ee 100644
--- a/src/contrib/minizip/ioapi.h
+++ b/src/contrib/minizip/ioapi.h
@@ -21,7 +21,7 @@
 #ifndef _ZLIBIOAPI64_H
 #define _ZLIBIOAPI64_H
 
-#if (!defined(_WIN32)) && (!defined(WIN32)) && (!defined(__APPLE__))
+#if (!defined(_WIN32)) && (!defined(WIN32)) && (!defined(__APPLE__)) && (!defined(__ANDROID__))
 
   // Linux needs this to support file operation on files larger then 4+GB
   // But might need better if/def to select just the platforms that needs them.
diff --git a/src/contrib/minizip/miniunz.c b/src/contrib/minizip/miniunz.c
index 3d65401..07bbe21 100644
--- a/src/contrib/minizip/miniunz.c
+++ b/src/contrib/minizip/miniunz.c
@@ -12,7 +12,7 @@
          Copyright (C) 2009-2010 Mathias Svensson ( http://result42.com )
 */
 
-#if (!defined(_WIN32)) && (!defined(WIN32)) && (!defined(__APPLE__))
+#if (!defined(_WIN32)) && (!defined(WIN32)) && (!defined(__APPLE__)) && (!defined(__ANDROID__))
         #ifndef __USE_FILE_OFFSET64
                 #define __USE_FILE_OFFSET64
         #endif
diff --git a/src/contrib/minizip/minizip.c b/src/contrib/minizip/minizip.c
index 4288962..a891af0 100644
--- a/src/contrib/minizip/minizip.c
+++ b/src/contrib/minizip/minizip.c
@@ -13,7 +13,7 @@
 */
 
 
-#if (!defined(_WIN32)) && (!defined(WIN32)) && (!defined(__APPLE__))
+#if (!defined(_WIN32)) && (!defined(WIN32)) && (!defined(__APPLE__)) && (!defined(__ANDROID__))
         #ifndef __USE_FILE_OFFSET64
                 #define __USE_FILE_OFFSET64
         #endif
@@ -28,7 +28,7 @@
         #endif
 #endif
 
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(__ANDROID__)
 // In darwin and perhaps other BSD variants off_t is a 64 bit value, hence no need for specific 64 bit functions
 #define FOPEN_FUNC(filename, mode) fopen(filename, mode)
 #define FTELLO_FUNC(stream) ftello(stream)
-- 
2.17.1

